astropy/utils/shapes.py

```diff
@@ -385,7 +410,7 @@ def check_broadcast(*shapes):
     return tuple(full_shape[::-1])
 
 
-def unbroadcast(array):
+def unbroadcast(array: NDArray) -> NDArray:
```

Comment:
NDArray can be parametrized by the dtype, which is preserved in this function.

```
def unbroadcast(array: NDArray[DT]) -> NDArray[DT]:
```

Reply:
That sounds simple enough.